cpp/include/raft/core/resources.hpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "resource/resource_types.hpp"
+#include <algorithm>
+#include <mutex>
+#include <raft/core/logger.hpp>
+#include <string>
+#include <vector>
+
+namespace raft {
+
+template <typename T>
+using pair_res = std::pair<resource::resource_type, std::shared_ptr<T>>;
```

Comment:
These should be in the class body, and not at the namespace level so these alias names don't leak